sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/repository/alternative/SqlEntityRepository.java

```diff
@@ -0,0 +1,166 @@
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.sodalite.server.repository.alternative;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.icgc.dcc.sodalite.server.utils.JsonUtils;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.Update;
+import org.skife.jdbi.v2.util.StringMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import lombok.SneakyThrows;
+import lombok.val;
+
+class SqlEntityRepository extends EntityRepository {
+
+  @Autowired
+  Handle h;
+  boolean parentsExist;
+
+  Update create;
+  Update update;
+  Update delete;
+
+  List<String> fields;
+  private String tableName;
+  private String parentIdField;
+
+  SqlEntityRepository(Handle h, String tableName, String parentIdField, List<String> fields) {
+    this.h = h;
+    this.tableName = tableName;
+    this.parentIdField = parentIdField;
+    this.fields = fields;
+    val insert = sqlCmdCreate(tableName, parentIdField, fields);
+    create = h.createStatement(insert);
+  }
+
+  private boolean parentsExist() {
+    return parentIdField == null;
+  }
+
+  String sqlCmdCreate(String tableName, String parentIdField, List<String> fields) {
+    String insert = "INSERT INTO " + tableName + " (id, studyId, ";
+    if (parentsExist()) {
+      insert += parentIdField;
+    }
+    insert += fieldList() + ") VALUES (?, ?";
+    if (parentsExist) {
+      insert += " ,?";
+    }
+    insert += bindings() + ")";
+    return insert;
+  }
+
+  @Override
+  public void create(String id, String parentId, String studyId, JsonNode data) {
+    int i = 0;
+    this.create.bind(i++, id);
+    this.create.bind(i++, studyId);
+
+    if (parentsExist) {
+      this.create.bind(i++, parentId);
+    }
+
+    for (val f : fields) {
+      this.create.bind(i++, data.get(f).asText());
+    }
+
+    create.execute();
+
+  }
+
+  private String bindings() {
+    String s = "";
+    for (int i = 1; i <= fields.size(); i++) {
+      s += " ,?";
+    }
+    return s;
+  }
+
+  private String fieldList() {
+    return fields.stream().collect(Collectors.joining(","));
+  }
+
+  @SneakyThrows
+  @Override
+  public JsonNode read(String id) {
+    val rs = h.select("select * from " + tableName + " WHERE ID = " + id);
+    val row = rs.get(0);
+
+    String json = "{";
+    if (parentsExist) {
+      json += JsonUtils.jsonResponse(parentIdField, row.get(parentIdField).toString());
+    }
+    for (val f : fields) {
+      json += JsonUtils.jsonResponse(f, row.get(f).toString());
+    }
+    json += "}";
+    return JsonUtils.getTree(json);
+  }
+
+  @Override
+  public void update(String id, JsonNode data) {
+    String sql = "UPDATE " + tableName + "SET ";
```

Comment:
Two problems here. First is that you are mutating a `String` several times. Second is that this is really loopy and hard to read as you loop and construct a string over several lines.

You should aim to get all your inputs ready and then add them to a sql template. That way your intention is easier to see as well as the reduced loopyness makes it less prone to developer error. 

```java
import static org.icgc.dcc.common.core.util.Joiners.COMMA;
.
.
.
  public void update(String id, JsonNode data) {
    val list = fields.stream().map(f -> format("%s=%s", f, data.get(f).asText())).collect(toList());
    val sql = format("UPDATE %s SET %s WHERE id=?", tableName, COMMA.join(list));
    h.execute(sql, id);
  }
```